raiden-cli/src/routes/userDeposit.ts

```diff
@@ -0,0 +1,151 @@
+}
+
+function getPlannedWithdrawAmount(request: Request): string | undefined {
+  return request.body.planned_withdraw_amount?.toString();
+}
+
+function getWithdrawAmount(request: Request): string | undefined {
+  return request.body.withdraw_amount?.toString();
+}
+
+function shouldDeposit(request: Request): boolean {
+  return (
+    getTotalDepositAmount(request) !== undefined &&
+    getPlannedWithdrawAmount(request) === undefined &&
+    getWithdrawAmount(request) === undefined
+  );
+}
+
+function shouldPlanWithdraw(request: Request): boolean {
+  return (
+    getTotalDepositAmount(request) === undefined &&
+    getPlannedWithdrawAmount(request) !== undefined &&
+    getWithdrawAmount(request) === undefined
+  );
+}
+
+function shouldWithdraw(request: Request): boolean {
+  return (
+    getTotalDepositAmount(request) === undefined &&
+    getPlannedWithdrawAmount(request) === undefined &&
+    getWithdrawAmount(request) !== undefined
+  );
+}
+
+async function deposit(this: Cli, totalDeposit: string): Promise<DepositResponse> {
+  const totalDepositTarget = parseAsBigNumber(totalDeposit);
+  const currentTotalDeposit = await this.raiden.getUDCCapacity();
+  const depositDifference = totalDepositTarget.sub(currentTotalDeposit);
+
+  if (depositDifference.lte(Zero)) throw TotalDepositTooLowError;
+
+  const transactionHash = await this.raiden.depositToUDC(depositDifference);
+  return { transaction_hash: transactionHash };
+}
+
+async function planWithdraw(this: Cli, amount: string): Promise<PlanWithdrawResponse> {
+  const transactionHash = await this.raiden.planUDCWithdraw(amount);
+  const withdrawPlan = await this.raiden.getUDCWithdrawPlan();
+  return {
+    transaction_hash: transactionHash,
+    planned_withdraw_block_number: withdrawPlan!.block, // The plan must exists here.
+  };
+}
+
+async function withdraw(this: Cli, amount: string): Promise<WithdrawResponse> {
+  const transactionHash = await this.raiden.withdrawFromUDC(amount);
+  return { transaction_hash: transactionHash };
+}
+
+async function determineAndExecuteRequestedInteraction(
+  this: Cli,
+  request: Request,
+): Promise<DepositResponse | PlanWithdrawResponse | WithdrawResponse> {
+  if (shouldDeposit(request)) {
+    const totalDeposit = getTotalDepositAmount(request)!; // Can't be undefined here.
+    return await deposit.call(this, totalDeposit);
+  } else if (shouldPlanWithdraw(request)) {
+    const amount = getPlannedWithdrawAmount(request)!; // Can't be undefined here.
+    return await planWithdraw.call(this, amount);
+  } else if (shouldWithdraw(request)) {
+    const amount = getWithdrawAmount(request)!; // Can't be undefined here.
+    return await withdraw.call(this, amount);
+  } else {
+    throw MalformedRequestFormat;
+  }
+}
+
+async function getUDCWithdrawPlan(
+  this: Cli,
+  _request: Request,
+  response: Response,
+): Promise<void> {
+  const withdrawPlan = await this.raiden.getUDCWithdrawPlan();
```

Comment:
You need to call `toString` on the `amount`, it´s serializing the BigNumber as an object

Reply:
✔️